jetty-websocket/websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/WebSocketComponents.java

```diff
@@ -43,41 +46,54 @@ public WebSocketComponents()
     public WebSocketComponents(WebSocketExtensionRegistry extensionRegistry, DecoratedObjectFactory objectFactory,
                                ByteBufferPool bufferPool, InflaterPool inflaterPool, DeflaterPool deflaterPool)
     {
-        this.extensionRegistry = (extensionRegistry == null) ? new WebSocketExtensionRegistry() : extensionRegistry;
-        this.objectFactory = (objectFactory == null) ? new DecoratedObjectFactory() : objectFactory;
-        this.bufferPool = (bufferPool == null) ? new MappedByteBufferPool() : bufferPool;
-        this.inflaterPool = (inflaterPool == null) ? new InflaterPool(CompressionPool.DEFAULT_CAPACITY, true) : inflaterPool;
-        this.deflaterPool = (deflaterPool == null) ? new DeflaterPool(CompressionPool.DEFAULT_CAPACITY, Deflater.DEFAULT_COMPRESSION, true) : deflaterPool;
+        this (extensionRegistry, objectFactory, bufferPool, inflaterPool, deflaterPool, null);
+    }
+
+    public WebSocketComponents(WebSocketExtensionRegistry extensionRegistry, DecoratedObjectFactory objectFactory,
+                               ByteBufferPool bufferPool, InflaterPool inflaterPool, DeflaterPool deflaterPool, Executor executor)
+    {
+        _extensionRegistry = (extensionRegistry == null) ? new WebSocketExtensionRegistry() : extensionRegistry;
+        _objectFactory = (objectFactory == null) ? new DecoratedObjectFactory() : objectFactory;
+        _bufferPool = (bufferPool == null) ? new MappedByteBufferPool() : bufferPool;
+        _inflaterPool = (inflaterPool == null) ? new InflaterPool(CompressionPool.DEFAULT_CAPACITY, true) : inflaterPool;
+        _deflaterPool = (deflaterPool == null) ? new DeflaterPool(CompressionPool.DEFAULT_CAPACITY, Deflater.DEFAULT_COMPRESSION, true) : deflaterPool;
+        _executor = (executor == null) ? new QueuedThreadPool() : executor;
```

Comment:
rather than create a whole new pool here, I think it would be better to use an `Executor` that just starts a new thread for each call to `execute` - so if you don't pass in a thread pool then you don't get a thread pool.

I know we do new QTP in other places... but I'm questioning if that is still the right thing to do.

Reply:
If you start a `WebSocketClient` by default you will use this `executor` for the `HttpClient`. I don't think we'd really want this strategy of starting a new thread every time for the client.

Reply:
OK